### PR TITLE
Issue/957 - Updated text for vagrant installations

### PR DESCRIPTION
**Pull request by ryan-lynch**
The vagrant installation instructed the user that they will be prompted for configuration settings, but this is not the case usually with vagrant installations, the args are taken from the vagrant setup. Tweaked the text here so it is clear.

Run into an issue where only the first 5 args were saved to config.json - copied optimist.argv code to the top of the install script to fix this.

### CODE REVIEW

File: install.js [hunk 1]
```diff
@@ -414,7 +422,22 @@ var steps = [
   },
   // all done
   function finalize (next) {
-    console.log("Installation complete.\nRun the command 'node server' (or 'foreman start' if using heroku toolbelt) to start your instance.");
+    var passedArgs = process.argv.slice(2);
```
Comment (dennis-learningpool) [inline install.js]: lines 425 - 434 are calling out to be refactored into their own function.

File: install.js [hunk 2]
```diff
@@ -24,6 +29,8 @@ var app = builder();
 var masterTenant = false;
 var superUser = false;
 
+var isVagrantInstallation = false;
```
Comment (dennis-learningpool) [inline install.js]: this var could be initialized here with the function suggested in my comment below.

File: install.js [hunk 3]
```diff
@@ -414,7 +422,22 @@ var steps = [
...
+    for (var i = 2; i < passedArgs.length; i++) {
+      if (passedArgs[i] == undefined) {
+        isVagrantInstallation = false;
+        break;
+      } else {
+        isVagrantInstallation = true;
+      }
+    }
+
+    if (isVagrantInstallation) {
```
Comment (brian-learningpool) [inline install.js]: You're looping over each argument here and setting `isVagrantInstallation = true` multiple times.  It's probably better to just do something like:

`isVagrantInstallation = process.argv.length == 2 ? false : true;`

Also, can you rename this variable to `isVagrant`, for brevity?  The 'installation' part can be inferred from the fact that you're in the installation script (install.js).

You could also move this check further up the script (at the start) so other pieces of the code can know if if it's a Vagrant install.

Comment (ryan-lynch) [inline install.js]: @brian-learningpool see most recent commit, I have taken yours and Dennis' points on board.

File: install.js [hunk 4]
```diff
@@ -24,6 +29,12 @@ var app = builder();
 var masterTenant = false;
 var superUser = false;
 
+var isVagrant = function () {
+  if (process.argv.length != 2) {
+      return true;
+    }
```
Comment (dennis-learningpool) [inline install.js]: minor indentation issue here. also, an explicit `return false;` would be a bit clearer:

````
if (process.argv.length != 2) {
      return true;
}

return false;
````

File: install.js [hunk 5]
```diff
@@ -24,6 +29,12 @@ var app = builder();
 var masterTenant = false;
 var superUser = false;
 
+var isVagrant = function () {
+  if (process.argv.length != 2) {
```
Comment (dennis-learningpool) [inline install.js]: I'd also be inclined to use `> 2` over `!= 2`

**Comment by dennis-learningpool** [comment]
+1

**Comment by brian-learningpool** [comment]
+1

**Pull request merged**